### ISSUE TITLE
everytime the app will crash because the retreived data is empty

### ISSUE THREAD

**Issue by AlanKong99** [googlecodelabs/android-paging#95]
The code below is extracted from GithubRemoteMediator. It did not deal with exceptions. Actually, you need to because everytime if your data retreived from database or network is empty, your app will crash. For example, if I tried to search 'abcsde', the app will crash because the response is empty. (I used the final version of this codelabs doing this search)

```
LoadType.PREPEND -> {
                ...
                val remoteKeys = getRemoteKeyForFirstItem(state)
                        ?:throw InvalidObjectException("Remote key and the prevKey should not be null")
                ...
            }
            LoadType.APPEND -> {
                ...
                if (remoteKeys == null || remoteKeys.nextKey == null) {
                    throw InvalidObjectException("Remote key should not be null for $loadType")
                }
                ...
            }
```

Most importantly, the crash is very annoying. So we need to dealt with our exceptions. And you could do it like the code below: 

```
 LoadType.PREPEND -> {

                val remoteKeys = getRemoteKeyForFirstItem(state)
                    ?: return MediatorResult.Error(InvalidObjectException("Remote key and the prevKey should not be null"))// I do not know wehther this is a good description, maybe you could replace it with a new 
              ....
            }
            LoadType.APPEND -> {
                ...
                if (remoteKeys?.nextKey == null) {
                    return MediatorResult.Error(InvalidObjectException("Remote key should not be null for $loadType")) // I do not know wehther this is a good description
                }
                ...
            }
```

**Comment by yangjunfei-lotus** [comment]
Someone addressed this issue?

**Comment by AdamSHurwitz** [comment]
@yangjunfei-lotus, I've created the pull request, [fix: RemoteMediator #112][pr112], in order to resolve this issue by disabling the Prepend load which is not being used, and to set a default value for the first load of the Append load type.

[pr112]: https://github.com/googlecodelabs/android-paging/pull/112

**Comment by florina-muntenescu** [comment]
The implementation of `LoadType.PREPEND` and `LoadType.APEND` changed meanwhile. This should have been fixed. Let me know if it didn't
